src/main/scala/regmapper/RegisterCrossing.scala

```diff
@@ -7,129 +7,167 @@ import chisel3.util.{Irrevocable}
-    val request_valid  = Bool(INPUT)
-    val response_ready = Bool(INPUT)
-    val busy           = Bool(OUTPUT)
+    val bypass                  = Bool(INPUT)
+    val master_request_valid    = Bool(INPUT)
+    val master_request_ready    = Bool(OUTPUT)
+    val master_response_valid   = Bool(OUTPUT)
+    val master_response_ready   = Bool(INPUT)
+    val crossing_request_valid  = Bool(OUTPUT)
+    val crossing_request_ready  = Bool(INPUT)
+    // ... no crossing_response_ready; we are always ready
   }
 
-  val busy = RegInit(Bool(false))
-  when (io.progress) {
-    busy := Mux(busy, !io.response_ready, io.request_valid)
+  val busy   = RegInit(Bool(false))
+  val bypass = Reg(Bool())
+
+  when (io.crossing_request_ready || Mux(busy, bypass, io.bypass)) {
+    busy := Mux(busy, !io.master_response_ready, io.master_request_valid)
+  }
+
+  when (io.master_request_valid && io.master_request_ready) {
+    bypass := io.bypass
   }
-  io.busy := busy
+
+  io.crossing_request_valid := io.master_request_valid && !io.bypass && !busy
+  io.master_request_ready   := (io.bypass || io.crossing_request_ready) && !busy
+  io.master_response_valid  := (bypass    || io.crossing_request_ready) &&  busy
+}
+
+class RegisterCrossingAssertion extends Module {
+  val io = new Bundle {
+    val master_bypass = Bool(INPUT)
+    val slave_reset = Bool(INPUT)
+  }
+
+  assert (io.master_bypass || !io.slave_reset)
 }
 
 // RegField should support connecting to one of these
 class RegisterWriteIO[T <: Data](gen: T) extends Bundle {
-  val request  = Decoupled(gen).flip()
+  val request  = Decoupled(gen).flip
   val response = Irrevocable(Bool()) // ignore .bits
 }
 
-// To turn on/off a domain:
-//  1. lower allow on the other side
-//  2. wait for inflight traffic to resolve
-//  3. assert reset in the domain
-//  4. turn off the domain
-//  5. turn on the domain
-//  6. deassert reset in the domain
-//  7. raise allow on the other side
+// To turn off=>on a domain:
+//  A. To turn disable the master domain
+//    1. wait for all inflight traffic to resolve
+//    2. assert master reset
+//    3. (optional) stop the master clock
+//    --- YOU MAY NOT TURN OFF POWER ---
+//    4. re-enable the clock
+//    5. deassert reset
+//  B. To turn off the slave domain
+//    1. assert bypass
+//    2. wait for inflight traffic to resolve
+//    3. assert slave reset
+//    4. (optional) stop the slave clock
+//    --- YOU MAT NOT TURN OFF POWER ---
+//    5. re-enable the clock
+//    6. deassert reset
+//    7. deassert bypass
+//
+// If you need to cut power, use something that support isolation gates.
 
 class RegisterWriteCrossingIO[T <: Data](gen: T) extends Bundle {
   // Master clock domain
   val master_clock   = Clock(INPUT)
   val master_reset   = Bool(INPUT)
-  val master_allow   = Bool(INPUT) // actually wait for the slave
   val master_port    = new RegisterWriteIO(gen)
+  // Bypass requests from the master to be noops
+  val master_bypass  = Bool(INPUT)
```

Comment:
One thing that might be good to comment...

If I do a write while the master_bypass is asserted, does it have any effect when master_bypass is de-asserted? 

In other words, on the slave side there is a register with value 0. I assert master_bypass, then do a write with value 0xFF. If I do a read with master_bypass, I just get 0 (right?) And if I de-assert master_bypass, the register still is 0, the value 0xFF is completely forgotten.

I'm pretty sure this is what you coded & intended, but might not be immediately obvious to someone deciding to use this.